### PR TITLE
Tweak copy re Stack Overflow / Server Fault / other SE network sites

### LINKED ISSUE #38821: Debugging task links to Stack Overflow (not Server Fault)
**This is a Bug Report**

**Problem:**

https://kubernetes.io/docs/tasks/debug/ links to Stack Overflow. It'd be better to link to Server Fault, because typical Kubernetes questions raised on Stack Overflow are liable to be closed as being off-topic for that site.

(the Stack Exchange Network operates both websites)

**Proposed Solution:**
Edit https://kubernetes.io/docs/tasks/debug/ to write about Server Fault / `serverfault.com` instead of Stack Overflow / `stackoverflow.com`

**Page to Update:**
https://kubernetes.io/docs/tasks/debug/

**Additional Information:**
/language en
/priority backlog
/help
/triage accepted

### PR DESCRIPTION
**Pull request by MaxHorstmann**
Fixes #38821

Clarify where to ask questions on Stack Overflow, Server Fault, other Stack Exchange sites.

### CODE REVIEW

**Comment by tengqm** [comment]
/lgtm
Thanks.

**Review by divya-mohan0209 — COMMENTED** [review]
From the issue description, it felt that Stack Overflow needed to be done away with as opposed to adding Server Fault being added. 
Deferring to @sftim since I'm not sure if this was in alignment with his vision.

**Comment by divya-mohan0209** [comment]
Also, I am not sure if your fork is in sync with the website since I cannot see a netlify preview. Please ensure you sync it since this typically happens when the fork is outdated.

**Comment by MaxHorstmann** [comment]
Right that's what the original issue said, but it's not quite correct that SO is always off-topic for Kubernetes questions. My copy suggestion clarifies what kind of question should be asked where. Source: I used to work for Stack Overflow.

**Comment by MaxHorstmann** [comment]
> Also, I am not sure if your fork is in sync with the website since I cannot see a netlify preview. Please ensure you sync it since this typically happens when the fork is outdated.

I synced the fork right before creating this PR. Is there a way to trigger netlify manually?

**Comment by MaxHorstmann** [comment]
Sync'ed the fork again, now there's a netlify preview. Can someone re-lgtm please?

**Comment by tengqm** [comment]
/lgtm
again.

File: content/en/docs/tasks/debug/_index.md [hunk 1]
```diff
@@ -43,14 +43,31 @@ and command-line interfaces (CLIs), such as [`kubectl`](/docs/reference/kubectl/
 
 ## Help! My question isn't covered!  I need help now!
 
-### Stack Overflow
+### Stack Overflow / Stack Exchange
```
Comment (divya-mohan0209) [inline content/en/docs/tasks/debug/_index.md:46]: I know Stack Exchange manages both of them, so shouldn't this be the title instead?
```suggestion
### Stack Overflow / Server Fault?
```

**Review by divya-mohan0209 — COMMENTED** [review]
Just a clarification

File: content/en/docs/tasks/debug/_index.md (hunk 1 above)
Comment (MaxHorstmann) [reply content/en/docs/tasks/debug/_index.md:46]: "Stack Exchange" technically refers to *all* the network sites, including Stack Overflow, Server Fault, and the other smaller sites mentioned in the paragraph (DevOps, Infosec...). So technically, the section title could be just "Stack Exchange"; however, Stack Overflow is by far the largest and best-known of these sites, and some people might not be familiar with "Stack Exchange" at all, hence "Stack Overflow / Stack Exchange". 

I don't feel super strongly about this, but would suggest to keep as-is.

**Review by Arhell — COMMENTED** [review]
/lgtm

File: content/en/docs/tasks/debug/_index.md [hunk 2]
```diff
@@ -43,14 +43,31 @@ and command-line interfaces (CLIs), such as [`kubectl`](/docs/reference/kubectl/
...
 
-Someone else from the community may have already asked a similar question or may
-be able to help with your problem. The Kubernetes team will also monitor
+Kubernetes questions related to *software development* can be asked on 
+[Stack Overflow](https://stackoverflow.com/questions/tagged/kubernetes).  
+
+Kubernetes questions related to *configuration* can be asked on
```
Comment (sftim) [inline content/en/docs/tasks/debug/_index.md:51]:
```suggestion
If you have Kubernetes questions related to *cluster management* or *configuration*,
you can ask those on
```

File: content/en/docs/tasks/debug/_index.md [hunk 3]
```diff
@@ -43,14 +43,31 @@ and command-line interfaces (CLIs), such as [`kubectl`](/docs/reference/kubectl/
 
 ## Help! My question isn't covered!  I need help now!
 
-### Stack Overflow
+### Stack Overflow / Stack Exchange
 
-Someone else from the community may have already asked a similar question or may
-be able to help with your problem. The Kubernetes team will also monitor
+Kubernetes questions related to *software development* can be asked on 
+[Stack Overflow](https://stackoverflow.com/questions/tagged/kubernetes).
```
Comment (sftim) [inline content/en/docs/tasks/debug/_index.md:49]:
```suggestion
If you have questions related to *software development* for your containerized app,
you can ask those on
```

File: content/en/docs/tasks/debug/_index.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/tasks/debug/_index.md:46]: Seems reasonable.
Aside:
Setting a fragment identifier would be slightly better in some ways:
```suggestion
### Stack Exchange, Stack Overflow, or Server Fault {#stack-exchange}
```

**Review by sftim — COMMENTED** [review]
Do you prefer these tweaks? We try to write in the active voice.

If you'd like to make a case for leaving things just as they are, that's OK. Reply to say so and a contributor will approve it as-is (I'm willing to).

**Comment by MaxHorstmann** [comment]
Agreed with those tweaks, thank you!

**Pull request merged**